sources/vcdMaker/inc/CliMaker.h

```diff
@@ -78,7 +79,7 @@ namespace CLI
         private:
 
             /// Valid timebases constraint.
-            TCLAP::ValuesConstraint<std::string> m_AllowedTimebases{SIGNAL::Signal::TIME_UNITS};
+            TCLAP::ValuesConstraint<std::string> m_AllowedTimebases{const_cast<std::vector<std::string>&>(TIME::Unit::GetInstance().GetTimeUnits())};
```

Comment:
W0000t? `const_cast` means something os grossly wrong. What happened?

Reply:
Oh I see, Tclap is crazy. There was even a note in `Signal` class about that. :-D I would say we should work-around it in this class - define member `vector` just to hold copy of what is in `TIME::Unit` class (with the same note added) and use it here in init.
```cpp
std::vector<std::string> m_AllowedTimebasesTclap{TIME::Unit::GetInstance().GetTimeUnits()};
TCLAP::ValuesConstraint<std::string> m_AllowedTimebases{m_AllowedTimebasesTclap};
```
Please also consider using `std::array` (as pointed out above) and all-static members (not singleton). Then it would look like this:
```cpp
std::vector<std::string> m_AllowedTimebasesTclap{TIME::Unit::GetTimeUnits().cbegin(),
                                                 TIME::Unit::GetTimeUnits().cend()};
```
We may consider using different library for argument parsing in future.

Reply:
I like the trick. I wouldn't give up Tclap just because of it. I really enjoy the output it provides. Well, nothing is perfect after all.

Reply:
But I also never say 'no' :)